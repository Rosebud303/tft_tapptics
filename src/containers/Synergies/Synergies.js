import React, { Component } from 'react';
import ReactModal from 'react-modal';
import ReactTooltip from 'react-tooltip';


export class Synergies extends Component {
  constructor() {
    super();
    this.state = {
      showModal: false,
      currentChoice: { attributes: { thumbnail: '', name: '', tier_info: []} }
    }
  }

  handleOpenModal = (e) => {
    const { id } = e.target
    let currentChoice = this.props.synergies.find(syn => syn.id == id);
    this.setState({showModal: true, currentChoice})
  }

  handleCloseModal = () => {
    this.setState({showModal: false, currentChoice: { attributes: { thumbnail: '', name: '', tier_info: [] } } })
  }
  
  render() {
    const displaySynergies = this.props.synergies.map(syn => (
     <article key={syn.id} id={syn.id} onClick={(e) => this.handleOpenModal(e)} className="synergy-article">
       <img onClick={(e) => this.handleOpenModal(e)} id={syn.id} src={syn.attributes.thumbnail} />
       <h3 id={syn.id} onClick={(e) => this.handleOpenModal(e)} >{syn.attributes.name}</h3>
       <p id={syn.id} onClick={(e) => this.handleOpenModal(e)} >{syn.attributes.summary}</p>
     </article>
    ))

    const matchingChamps = this.props.champions.reduce((acc, champion) => {
      champion.attributes.origin_class_types.map(type => {
        if (!acc.includes(type.name) && type.name == this.state.currentChoice.attributes.name) {
          acc.push(champion)
        }
      });
      return acc
      }, []);

    const displayModal = (
      <article className="display-synergy-modal">
      <button onClick={this.handleCloseModal}>X</button>
      <div className="syngery-info">
        <img src={this.state.currentChoice.attributes.thumbnail} alt={this.state.currentChoice.attributes.name} />
        <h3>{this.state.currentChoice.attributes.name}</h3>
        {this.state.currentChoice.attributes.tier_info.map(tier => (<p>{tier}</p>))}
      </div>
      <div className="champ-syngery-matches">
        {
          matchingChamps.map(champ => (
          <div>
            <a data-tip data-for={champ.attributes.name}>
              <img src={champ.attributes.champion_thumbnail} alt={champ.attributes.name} />
            </a>
            <ReactTooltip id={champ.attributes.name} type='dark' effect="float" className="modal-tooltip">
                <div>{champ.attributes.name}</div>
                {
                  champ.attributes.origin_class_types.map(type => (
                    <div>{type.name}</div>
                  ))
                }
              </ReactTooltip>
          </div>
          ))
        }
      </div>
      </article>
    )

    return (
      <div className="Synergies">
        <h3>Classes & Origins</h3>
        <section className="display-synergies">
        {displaySynergies}
        </section>
        <ReactModal
           isOpen={this.state.showModal}
           className="Modal"
           id="Synergy-modal"
           contentLabel="Synergy Modal">
           <div className="modal-display-synergies">
            { this.state.showModal && displayModal }
          </div>
        </ReactModal>
      </div>
    )
  }
}