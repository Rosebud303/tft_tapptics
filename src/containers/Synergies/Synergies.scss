.Synergies {
  background-image: linear-gradient($fade1, $fade2), url('../../assets/images/synergy.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  height: 90vh;
  overflow: scroll;
  padding: $page-padding;

  h3 {
    color: $body-font-color;
    font-size: 1.5rem;
    margin: 0;
  }

  .display-synergies {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;

    img {
      margin: 1rem;
      width: 2rem;
    }
  }
}

.synergy-article {
  background-color: $modal-overlay;
  border: 1px solid $medium-bg;
  color: #83806d;
  font-size: .9rem;
  margin: 1rem;
  padding: .5rem;
  text-align: center;
  width: 10rem;
  transition: .5s;
  &:hover {
    cursor: pointer;
    background-color: $card-hover-bg;
    box-shadow: $card-hover-shadow;

    h3 {
      text-decoration: underline;
    }
  }

  img {
    margin: auto;
  }
}

.ReactModal__Overlay--after-open {
  background-color: $modal-overlay !important;
}

#Synergy-modal {
  width: 50%;
}

.display-synergy-modal {
  display: grid;
  grid-template-areas: ". . . x"
  ". info info ."
  ". champ champ .";
  grid-template-columns: .5fr 2fr 2fr .5fr;
  text-align: center;

  img {
    margin: auto;
    width: 3rem;
  }

  button {
    grid-area: x;
  }

  .syngery-info {
    grid-area: info;
  }

  .champ-syngery-matches {
    grid-area: champ;
    display: flex;
    justify-content: space-between;
    margin: 1rem 0;
  }
}