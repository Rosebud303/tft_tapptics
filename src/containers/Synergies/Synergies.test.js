import React from 'react';
import { shallow } from 'enzyme';
import { Synergies } from './Synergies';

describe('Synergies', () => {

  let wrapper;
  let mockSynergies = [{ attributes: {name: 'Demon', thumbnail: '' } }];
  let mockChampions = [{ attributes: { name: 'Vayne', origin_class_types: ['Demon'] } }]

  beforeEach(() => {
    wrapper = shallow(<Synergies synergies={mockSynergies} champions={mockChampions} />);
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});