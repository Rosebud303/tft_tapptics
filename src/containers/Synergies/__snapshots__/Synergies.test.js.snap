// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Synergies should match the snapshot with all the data passed in 1`] = `
<div
  className="Synergies"
>
  <h3>
    Classes & Origins
  </h3>
  <section
    className="display-synergies"
  >
    <article
      className="synergy-article"
      onClick={[Function]}
    >
      <img
        onClick={[Function]}
        src=""
      />
      <h3
        onClick={[Function]}
      >
        Demon
      </h3>
      <p
        onClick={[Function]}
      />
    </article>
  </section>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="Modal"
    closeTimeoutMS={0}
    contentLabel="Synergy Modal"
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="modal-display-synergies"
    />
  </Modal>
</div>
`;
